onnxruntime/contrib_ops/cuda/decoder/fastertransformer_decoder_attention/decoder_masked_multihead_attention_impl.cu

```diff
@@ -314,9 +323,16 @@ __global__ void masked_multihead_attention_kernel(DecoderMaskedMultiheadAttentio
     // WARNING: ALL THE THREADS OF A WARP MUST ENTER!!!
     float qk = Qk_dot<T, THREADS_PER_KEY>::dot(q_vec, k_vec) * inv_sqrt_dh;
 
+    // This is a deviation from FasterTransformer kernel implementation
```

Comment:
The changes here and below are a deviation from FasterTransformer but tries to be consistent with our Attention kernels in how they use the mask_filter_value to avoid accuracy drop for PyTorch models

Reply:
Mask_filter_value is common for huggingface models and [Megatron models](https://github.com/NVIDIA/Megatron-LM/blob/8ce8256ff09373a275245aff3db68a3688218c44/megatron/fused_kernels/scaled_masked_softmax.h#L258). Parity is a priority for us (even at the cost of slight performance drop) so we shall apply it.

Reply:
Agree. Let's keep deviation for accuracy. FYI, in newer version, HF uses float.min as mask_filter_value instead of -10e4, so it is same essentially.